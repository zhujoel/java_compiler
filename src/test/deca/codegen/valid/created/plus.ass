; Test pour savoir si la pile est pleine
	TSTO #22
	BOV pile_pleine
	ADDSP #20
; start main program
; Class Declaration
; Code de la table des méthodes de Object
	LOAD #null, R0
	STORE R0, 14(GB)
	LOAD code.Object.equals, R0
	STORE R0, 15(GB)
; Main Function
; Declaration des variables
	LOAD #5, R2
	STORE R2, 16(GB)
	LOAD #2, R2
	STORE R2, 17(GB)
	LOAD #0x0.0p0, R0
	STORE R0, 18(GB)
	LOAD #0x0.0p0, R0
	STORE R0, 19(GB)
; Beginning of main instructions:
; +
	LOAD 16(GB), R2
	LOAD 17(GB), R3
	ADD R2, R3
	BOV debordement
; +
	LOAD #6, R2
	LOAD #1, R4
	ADD R2, R4
	BOV debordement
; +
	LOAD 16(GB), R2
	PUSH R4
	LOAD 17(GB), R4
	ADD R4, R2
	BOV debordement
	POP R4
	LOAD R2, R1
	WINT
; +
	LOAD 16(GB), R4
	PUSH R4
	LOAD 17(GB), R4
	ADD R4, R4
	BOV debordement
	POP R4
	STORE R4, 16(GB)
; +
	LOAD #5, R4
	PUSH R4
	LOAD #2, R4
	ADD R4, R4
	BOV debordement
	POP R4
	LOAD R4, R1
	WINT
	LOAD #5, R4
	FLOAT R4, R4
	STORE R4, 18(GB)
	LOAD #2, R4
	FLOAT R4, R4
	STORE R4, 19(GB)
; +
	LOAD 18(GB), R4
	PUSH R4
	LOAD 19(GB), R4
	ADD R4, R4
	BOV debordement
	POP R4
; +
	LOAD #0x1.8p2, R4
	PUSH R4
	LOAD #0x1.0p0, R4
	ADD R4, R4
	BOV debordement
	POP R4
; +
	LOAD 18(GB), R4
	PUSH R4
	LOAD 19(GB), R4
	ADD R4, R4
	BOV debordement
	POP R4
	LOAD R4, R1
	WINT
; +
	LOAD 18(GB), R4
	PUSH R4
	LOAD 19(GB), R4
	ADD R4, R4
	BOV debordement
	POP R4
	STORE R4, 18(GB)
; +
	LOAD #0x1.4p2, R4
	PUSH R4
	LOAD #0x1.0p1, R4
	ADD R4, R4
	BOV debordement
	POP R4
	LOAD R4, R1
	WINT
	HALT
code.Object.equals:
pile_pleine:
	WSTR "Erreur : Pile pleine."
	WNL
	ERROR
dereferencement_null:
	WSTR "Erreur : dereferencement null."
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : Tas plein."
	WNL
	ERROR
debordement:
	WSTR "Erreur : debordement durant une operation arithmétique"
	WNL
	ERROR
io_error:
	WSTR "Erreur : input ou output"
	WNL
	ERROR
; end main program
