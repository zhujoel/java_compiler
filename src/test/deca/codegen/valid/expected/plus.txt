7
7
7.00000e+00
7.00000e+00
